doc/adr/0011-tsp-analysis-api.md

```diff
@@ -178,100 +183,175 @@ sequenceDiagram
-    client->>server: POST /config/types/{typeId}
-    server->>client: 200: New Configuration
-    client->>client: Update list of existing Configuration
-    user->>client: Select experiment
-    user->>client: Open Configuration Selector UI for experiments
-    client->>client: Select Configuration (typeId, configId)
-    Note over client,server: Configuration can be assigned for different experiments
-    user->>client: Open trace
     client->>server: GET /experiments/{expUUID}/outputs
-    server->>client: 200: list of available outputs including In-And-Out outputs
-    client->>client: Refresh UI
+    server->>client: 200: List of Available Views
+    client->>client: Render "Available Views" view and config menu buttons
+    user->>client: Click config menu beside 'Flame Chart''
+    client->>server: GET /experiments/{expUUID}/outputs/{outputId}/configTypes {scope=view}
+    server->>client: 200: List of ConfigurationSourceTypes
+    client->>client: Render menu for each type
+    user->>client: Click on "Create Custom Flame Chart..."
+    client->>client: Open Dialog (Webview) to input parameters
+    user->>client: Fill in parameters'title' and 'filter' and click on 'Apply'
+    client->>server: POST experiments/{expUUID}/outputs/{outputId} {title, filter}
+    server->>server: Validate, create data provider, persist config for data provider
+    server->>client: 200: OutputDescriptor
+    client->>client: Refresh 'Available views' view
+    user->>client: User clicks on derived output
+    client->>client: Customized view opens
+ ```
+
+Notes: 
+- Data provider persists input parameters (configuraton) so that output is avaivable after closing of experiment or restart
+
+
+Example configuration source type:
+
+ConfigurationSourceType {
+    "name": Custom Flame Chart,
+    "description": Create instance of a custom flame based on input parameters,
+    "id": "custom.flame.charts.id",
+    "scope": "data provider",
+    "parameterDescriptors": [
+        {
+            "keyName": "title",
+            "description": "Provide a name of custom flame chart to be shown in UI",
+            "dataType": "string",
+            "isRequired": "true"
+        },
+        {
+            "keyName": "filter",
+            "description": "Provide a filter string according to filter language",
+            "dataType": "string",
+            "isRequired": "true"
+        }
+    ]
+}
+```
+
+Input data provider:
+
+```javascript
+OutputDescriptor {
+    "id": "flamechart.id",
+    "name": "Flame Chart",
+    "description": "Flame Chart description",
+    "type": "TIME_GRAPH"
+}
 ```
 
-#### Sequence: Delete configuration instance for an experiment
+`User input`:
+```javascript
+    {"title": "CPU 0-1 only", "filter": "cpu matches [0-1]"}
+```
+
+Resulting data provider:
+```javascript
+    OutputDescriptor {
+        id: string,
+        name: string,
+        description: string,
+        type: string,  // provider type
```

Comment:
Is there supposed to be a parentId?

Reply:
I think it makes sense. We can use it for grouping in the FE. If parentId is not present or -1 then it is considered root data provider.